app/src/main/java/es/situm/gettingstarted/drawbuilding/DrawBuildingActivity.java

```diff
@@ -83,23 +108,71 @@ void drawBuilding(Bitmap bitmap) {
         map.animateCamera(CameraUpdateFactory.newLatLngBounds(latLngBounds, 100));
     }
 
-    void fetchFirstFloorImage(Building building, Callback callback) {
+    void fetchFloorsImages(Building building, Callback callback) {
         SitumSdk.communicationManager().fetchFloorsFromBuilding(building.getIdentifier(), new Handler<Collection<Floor>>() {
             @Override
             public void onSuccess(Collection<Floor> floorsCollection) {
                 List<Floor> floors = new ArrayList<>(floorsCollection);
-                Floor firstFloor = floors.get(0);
-                SitumSdk.communicationManager().fetchMapFromFloor(firstFloor, new Handler<Bitmap>() {
+
+                // First instance a new LinkedHashMap to store all the floors in the same order they came [FIFO]
+                buildingFloors = new LinkedHashMap<>();
+                for(int i=floors.size()-1;i>=0;i--){
+                    Floor floor = floors.get(i);
+                    buildingFloors.put(floor.getIdentifier(), floor);
+                }
+
+                // Secondly, if the building has more than 3 floors, display the scroll marks to indicate it
+                if(buildingFloors.size() > 3){
+                    selectorMarkTop.setVisibility(View.VISIBLE);
+                    selectorMarkBottom.setVisibility(View.VISIBLE);
+                }
+
+                // Secondly instance the adapter of the selector
+                floorAdapter = new FloorAdapter(new ArrayList<>(buildingFloors.values()), floor -> {
+
+                    SitumSdk.communicationManager().fetchMapFromFloor(floor, new Handler<Bitmap>() {
+                        @Override
+                        public void onSuccess(Bitmap bitmap) {
+                            callback.onSuccess(bitmap);
+                            floorAdapter.select(floor);
```

Comment:
Entendo que o adapter é notificado de que cambiou a planta só cando xa se obtivo o bitmap. Eso quere dicir que cando a rede é lenta tocarás no botón para cambiar de planta e non sucederá nada ata 10 ou 15 segundos máis tarde?
Si é así, aquí vexo que hai 3 cousas que deberían terse en conta por separado:
1. O que sucede localmente que podemos controlar sin agardar pola imaxe.
2. O que sucede mentres se carga a imaxe (neste caso non hai loadings nin nada, pero puidera darse o caso e incluso non o descarto).
3. O que sucede posteriormente, cando por fin temos a imaxe.
Como o ves?